src/com/carolinarollergirls/scoreboard/json/JSONStateSnapshotter.java

```diff
@@ -28,14 +28,29 @@ public synchronized void sendUpdates(Map<String, Object> state, Set<String> chan
         if (state.get("ScoreBoard.CurrentPeriodNumber") != "0") {
             // If the jam has just ended or the score is now official, write out a file.
             if ((inJam && !bool(state.get("ScoreBoard.InJam")))
-                    || (!officialScore && bool(state.get("ScoreBoard.OfficialScore")))) {
+                    || (bool(state.get("ScoreBoard.OfficialScore"))
```

Comment:
A simpler way to do this would be to make this just `bool(state.get("ScoreBoard.OfficialScore")` without the state. We can then filter out the Clock changes if we want to.

Reply:
What's the point of splitting the conditions on which to write the file to two places? Or do I misunderstand the suggestion?

Reply:
My point is that rather than looking for the point that OfficialScore is changed and writing out only then, that we update the file continuously once the score is final.

Reply:
Which is what I am doing. It used to look for a change in the old code.

Reply:
Ah right, I'm not quite awake. I suggest not having the two clauses after this around locking and relevant updates.